amulet/deployer.py

```diff
@@ -136,6 +136,15 @@ def add(self, service_name,
         if placement is not None:
             service['to'] = placement
 
+        if constraints is None and 'JUJU_TEST_CONSTRAINTS' in os.environ:
```

Comment:
My $.02:

1. JUJU_TEST_CONSTRAINTS is a good name.
2. The constraints from the env var should always be applied if they exist. The test author's constraints should be applied over the top of them, so they take precedence in the case of a conflict.